civicrm_resources/civicrm_resources.module

```diff
@@ -19,38 +19,40 @@ function civicrm_resources_add_resources($extention_path, $req_files = []) {
   global $base_url;
 
   $files = civicrm_resources_file_scan_directory($extention_path, '/\.(css|js)$/i');
-
   $js_list = [];
   $css_list = [];
+  list($all_js, $all_css) = filter_files($files);
 
-  foreach ($files as $key => $file) {
-    if (!empty($req_files)) {
-      if (in_array('*.js', $req_files) && preg_match('/\.js$/', $file->filename)) {
-        $js_list[] = $file->uri;
-      }
-      else {
-        if (in_array($file->filename, $req_files) && preg_match('/\.js$/', $file->filename)) {
-          $js_list[] = $file->uri;
-        }
-      }
-      if (in_array('*.css', $req_files) && preg_match('/\.css$/', $file->filename)) {
-        $css_list[] = $file->uri;
-      }
-      else {
-        if (in_array($file->filename, $req_files) && preg_match('/\.css$/', $file->filename)) {
-          $css_list[] = $file->uri;
-        }
-      }
-    }
-    else {
-      if (preg_match('/\.css$/', $file->filename)) {
-        $css_list[] = $file->uri;
+  if (empty($req_files)) {
+    $req_files = ['*.css', '*.js'];
+  }
+
+  if (in_array('*.js', $req_files)) {
+    $js_list = get_uris($all_js);
+  }
+
+  if (in_array('*.css', $req_files)) {
+    $css_list = get_uris($all_css);
+  }
+
+  if (!in_array('*.css', $req_files) || !in_array('*.js', $req_files)) {
+    $resourceFilenameUri = array_map(function ($file) {
```

Comment:
I wonder if it wouldn't simply be easier to change `civicrm_resources_file_scan_directory` to return an array of arrays instead of objects. This way, it wouldn't be necessary to use `array_map` here to create this array

Reply:
Using array will be much better since the `array_column` in the following requires an array, making the `array_map` unnecessary.